contrib/regionTrimmer.py

```diff
@@ -45,114 +56,130 @@ def get_graph_bounds(graph):
         min(n[1] for n in nodes),
     )
 
+
 def get_graph_center_by_bounds(bounds):
     dx = bounds[0] - bounds[1]
     dy = bounds[2] - bounds[3]
-    return (dx / 2 + bounds[1], dy / 2 + bounds[3])
-
-def main(*args, **options):
-    if len(args) < 1:
-        logger.error('Missing region directory argument')
-        return
-    for path in args:
-        logger.info('Using regionset path: %s', path)
+    return (dx // 2 + bounds[1], dy // 2 + bounds[3])
+
+
+def trim_regions(graph, regions_path, dry_run=True, filter_func=lambda n: True):
+    regions = [
+        (n, get_region_file_from_node(regions_path, n))
+        for n in graph.nodes()
+        if filter_func(n)
+    ]
+    logger.info("Trimming regions: %s", ", ".join(x[1] for x in regions))
+    for n, region_file in regions:
+        graph.remove_node(n)
+        if dry_run is False:
+            unlink_file(region_file)
+
+
+def is_outside_main(center, main_section_bounds):
+    return center[0] <= main_section_bounds[0] and center[0] >= main_section_bounds[1] and \
+        center[1] <= main_section_bounds[2] and center[1] >= main_section_bounds[3]
+
+
+def is_outside_bounds(node, trim_center, trim_bounds):
+    return node[0] >= trim_center[0] + trim_bounds[0] or \
+        node[0] <= trim_center[0] - trim_bounds[0] or \
+        node[1] >= trim_center[1] + trim_bounds[1] or \
+        node[1] <= trim_center[1] - trim_bounds[1]
+
+def unlink_file(path):
+    try:
+        path.unlink()
+    except OSError as err:
+        logger.warn("Unable to delete file: %s", path)
```

Comment:
I think these here should be `logger.warning`,

```regionTrimmer.py:111: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead```

Reply:
Good point - fixed it now :+1: